### PR TITLE
NDEF reading and I2C connection

### LINKED ISSUE #29: Use Hardware spi
Esphome supports Hardware SPI why not use it?

the only thing you have to do is use the dedicated pins:
```
spi:
  clk_pin: D5
  mosi_pin: D7
  miso_pin: D6
```
then Esphome will use hardware SPI:
```
[12:00:12][C][spi:081]: SPI bus:
[12:00:12][C][spi:082]:   CLK Pin: GPIO14 (Mode: OUTPUT)
[12:00:12][C][spi:083]:   MISO Pin: GPIO12 (Mode: INPUT)
[12:00:12][C][spi:084]:   MOSI Pin: GPIO13 (Mode: OUTPUT)
[12:00:12][C][spi:085]:   Using HW SPI: YES
```
cs_pin can stay D3
buzzer can stay D8
but LED needs a new pin

### PR DESCRIPTION
**Pull request by jesserockz**
This PR is taking over from #33 

Also we have been looking at connecting via I2C as SPI was causing wifi issues.

Related issues:
- #34 
- #44 

To try this out you will need to change the wiring to match the schematic in this PR.
It also has a custom header file `ndef.h` which needs to be placed in your esphome config directory alongside the yaml file.

I am working on I2C and NDEF support for the PN532 directly in esphome/esphome#1302 but that is going to take time so using these libraries is a win for the meantime.

closes #29 as it moves to using I2C

### CODE REVIEW

**Comment by gadgetshed** [comment]
Got this running fine but the schemtic show buzzer on D7 when the yaml code has it on D3.  It's runnign a lot more stable than the spi version.  I have one of each version running side by side.

**Comment by jesserockz** [comment]
Thanks for noticing @gadgetshed I have updated the yaml. Not sure how that slipped in there.

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ The tag reader is a simple to build/use NFC tag reader, specially created for [H
 
 > I am selling a pre-built version, a DIY version (assembly required) or just the case (use with own components). Check it out on [my website](https://adonno-crafts.myshopify.com/).
 
-![Photos of the final product](https://raw.githubusercontent.com/adonno/tagreader/master/docs/cases.jpg)
```
Comment (balloob) [inline README.md:11]: Would be helpful if we get unrelated changes in a separate PR in case this PR takes a while to complete.

File: scripts/flash.bat [hunk 2]
```diff
@@ -0,0 +1,14 @@
+set mac=$(python -m esptool.py read_mac | grep -m 1 'MAC:' | grep -o '.\{8\}$' )
```
Comment (balloob) [inline scripts/flash.bat:1]: Same.

Comment (jesserockz) [reply scripts/flash.bat:1]: Yeah sorry this file was pushed in by accident

File: README.md (hunk 1 above)
Comment (jesserockz) [reply README.md:11]: #46

**Comment by jesserockz** [comment]
Closing as I2C support for PN532 has been merged into ESPHome `dev` branch (esphome/esphome#1302) and will be released with 1.16.0
